### PR TITLE
Merge mcmc_diagnostics, which includes recent updates from report.generation branch

### PR DESCRIPTION
**Pull request by alsnhll**
Summary of changes
-report.generation package includes many edits to correct/improve/standardize existing plots, and new functions to read in and plot other model output - including SEIR and outcome parameters and interventions, likelihoods, acceptance rates, seeding, and all model compartments
-filter_MC.R and its helper functions (in filter_MC_runner_funcs.R, functions.R) updated to save the intermediate chimeric files and track parameter acceptance decisions and step sizes,  so that progress of the MCMC can be tracked better. In addition parameter perturbation sd values are now only read from the config at the first step (when initial condition read) and kept in dataframe, instead of reading from config each step. 
-throughout R and python code more detailed commenting added and typos corrected

### CODE REVIEW

File: R/pkgs/inference/R/functions.R [hunk 1]
```diff
@@ -544,3 +546,195 @@ iterateAccept <- function(ll_ref, ll_new) {
   }
   return(FALSE)
 }
+
```
Comment (jkamins7) [inline R/pkgs/inference/R/functions.R:549]: Not for this PR, but do you mind splitting these off into a different file?

File: R/pkgs/inference/tests/testthat/test-perform_MCMC_step_copies.R [hunk 2]
```diff
@@ -2,7 +2,7 @@ context("perform_MCMC_step_copies")
 
 
 ##THESE TESTS CAN BE MADE MORE DETAILED...JUST MAKING PLACE HOLDERS
-test_that("MCMC step copies are correctly performed when we are not at the start of a block", {
+test_that("MCMC step copies (global) are correctly performed when we are not at the start of a block", {
```
Comment (jkamins7) [inline R/pkgs/inference/tests/testthat/test-perform_MCMC_step_copies.R:5]: This test needs to be updated to also do chimerics.  Again, not for this PR

File: R/pkgs/inference/tests/testthat/test-perform_MCMC_step_copies.R [hunk 3]
```diff
@@ -122,3 +122,125 @@ test_that("MCMC step copies are correctly performed when we are at the start of
 
 
 })
+
```
Comment (jkamins7) [inline R/pkgs/inference/tests/testthat/test-perform_MCMC_step_copies.R:125]: Oh, great. Ignore previous comment.

File: R/pkgs/report.generation/NAMESPACE [hunk 4]
```diff
@@ -36,27 +45,39 @@ export(make_scn_time_summary_table_withVent)
...
 export(plot_ts_incid_death_state_sample_allPdeath)
 export(plot_ts_incid_inf_state_sample)
+export(pretty_num)
+export(pretty_prop)
 export(print_pretty_date)
 export(print_pretty_date_short)
 export(read_file_of_type)
 export(reference_chunk)
 export(rt_estimates)
-importFrom(magrittr,"%>%")
```
Comment (jkamins7) [inline R/pkgs/report.generation/NAMESPACE:62]: I'm concerned that we lost this importFrom. I thought it was important?

File: R/pkgs/report.generation/R/setup_testing_environment.R [hunk 5]
```diff
@@ -0,0 +1,221 @@
+#' @export
+setup_testing_environment <- function(cf = "config.yml"){
```
Comment (jkamins7) [inline R/pkgs/report.generation/R/setup_testing_environment.R:2]: Didn't we delete this forever and ever and ever

File: R/pkgs/report.generation/tests/testthat/test-county_loading.R [hunk 6]
```diff
@@ -92,7 +92,7 @@ test_that("Simulation loading works", {
             scenario_labels = 'baseline',
             incl_geoids = included_geoids,
             inference=FALSE
-    ))+2
```
Comment (jkamins7) [inline R/pkgs/report.generation/tests/testthat/test-county_loading.R:95]: Why did this change?

File: local_install.R [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
+
 library(devtools)
   
+# To run if operating in the container
 initial.options <- commandArgs(trailingOnly = FALSE)
 file.arg.name <- "--file="
-script.name <- sub(file.arg.name, "", initial.options[grep(file.arg.name, initial.options)])
-
-pkg.dir <- paste0(dirname(script.name), "/R/pkgs/")
+script.name <- sub(file.arg.name, "", initial.options[grep(file.arg.name, initial.options)]) # get the name of this file, by looking for the option "--file" in the arguments that were used to start this R instance and getting the term that comes after
```
Comment (jkamins7) [inline local_install.R:8]: Generally to make code review easier, we prefer comments on their own lines.  Not an issue for this PR

File: minimal_interface.py [hunk 8]
```diff
@@ -38,7 +38,8 @@
 spatial_base_path = pathlib.Path(spatial_config["base_path"].get())
 scenario = scenario
 deathrate = deathrate
-stoch_traj_flag= stoch_traj_flag # Truthy: stochastic simulation, Falsy: determnistic mean of the binomial draws
+stoch_traj_flag= stoch_traj_flag # Truthy: stochastic simulation, Falsy: deterministic mean of the binomial draws
```
Comment (jkamins7) [inline minimal_interface.py:41]: Comment on own line

File: minimal_interface.py [hunk 9]
```diff
@@ -38,7 +38,8 @@
...
+
 nsim = 1
 interactive = False
 write_csv = False
```
Comment (jkamins7) [inline minimal_interface.py:45]: Did you mean to add the MCMC_diagnostics html file (couldn't comment on that file so commenting here)?

**Review by jkamins7 — COMMENTED** [review]
We can merge this as long as we commit to making some changes after.

File: minimal_interface.py (hunk 9 above)
Comment (alsnhll) [reply minimal_interface.py:45]: Yup, I wanted there to be an example report that people could look at without having to re-run it.

File: R/pkgs/report.generation/tests/testthat/test-county_loading.R (hunk 6 above)
Comment (alsnhll) [reply R/pkgs/report.generation/tests/testthat/test-county_loading.R:95]: I think this is a question for @juanderone ... I didn't edit this. maybe this is an accidental carryover from an older branch

File: R/pkgs/report.generation/R/setup_testing_environment.R (hunk 5 above)
Comment (alsnhll) [reply R/pkgs/report.generation/R/setup_testing_environment.R:2]: I think this is a question for @juanderone ... I didn't edit this. maybe this is an accidental carryover from an older branch

File: R/pkgs/report.generation/NAMESPACE (hunk 4 above)
Comment (alsnhll) [reply R/pkgs/report.generation/NAMESPACE:62]: I think this is a question for @juanderone ... I didn't edit this. maybe this is an accidental carryover from an older branch

File: R/pkgs/inference/R/functions.R (hunk 1 above)
Comment (alsnhll) [reply R/pkgs/inference/R/functions.R:549]: do you mean the stuff under the heading "Extra functions for MCMC diagnosis"? if so im happy to do that. they are almost-parallels of functions that are already in here, which is why I put them here, but no problem to move

**Pull request merged**